libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -933,6 +933,25 @@ string YulUtilFunctions::calldataArrayIndexAccessFunction(ArrayType const& _type
 	});
 }
 
+string YulUtilFunctions::calldataArrayIndexRangeAccess(ArrayType const& _type)
+{
+	solAssert(_type.dataStoredIn(DataLocation::CallData), "");
+	string functionName = "calldata_array_index_range_access_" + _type.identifier();
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(offset, length, startIndex, endIndex) -> offsetOut, lengthOut {
+				if gt(startIndex, endIndex) { invalid() }
+				if gt(endIndex, length) { invalid() }
```

Comment:
Instead of using ``invalid()`` here you should revert and transfer the revert reason strings used in the old code generation. You can have a look at the reverts in ``accessCalldataTailFunction`` for an example of this in Sol->Yul.

Reply:
I agree with this, updated and fixing tests now